File: app/src/main/java/com/example/apaga/ui/base/BaseExpandableRecyclerViewAdapter.java
package com.example.apaga.ui.base;

import android.view.ViewGroup;

import com.thoughtbot.expandablerecyclerview.ExpandableRecyclerViewAdapter;
import com.thoughtbot.expandablerecyclerview.models.ExpandableGroup;
import com.thoughtbot.expandablerecyclerview.viewholders.ChildViewHolder;
import com.thoughtbot.expandablerecyclerview.viewholders.GroupViewHolder;

import java.util.List;

public class BaseExpandableRecyclerViewAdapter <GVH extends GroupViewHolder, CVH extends ChildViewHolder>
        extends ExpandableRecyclerViewAdapter<GVH, CVH> {
    public BaseExpandableRecyclerViewAdapter(List<? extends ExpandableGroup> groups) {
        super(groups);
    }

    public void notifyGroupDataChanged() {
        expandableList.expandedGroupIndexes = new boolean[getGroups().size()];
        for (int i = 0; i < getGroups().size(); i++) {
            expandableList.expandedGroupIndexes[i] = false;
        }

    }
//    public void notifyChildDataChanged(int position) {
//        expandableList =
//        expandableList.expandedGroupIndexes = new boolean[getGroups().get(position).getItems().size()];
//        for (int i = 0; i < getGroups().get(position).getItems().size(); i++) {
//            expandableList.ge = false;
//        }
//        getGroups().get().getItems()
//    }

    @Override
    public GVH onCreateGroupViewHolder(ViewGroup parent, int viewType) {
        return null;
    }

    @Override
    public CVH onCreateChildViewHolder(ViewGroup parent, int viewType) {
        return null;
    }

    @Override
    public void onBindChildViewHolder(CVH holder, int flatPosition, ExpandableGroup group, int childIndex) {

    }

    @Override
    public void onBindGroupViewHolder(GVH holder, int flatPosition, ExpandableGroup group) {

    }
}

File: app/src/main/java/com/example/apaga/ui/base/BaseExpandableRecyclerViewAdapter.kt
package com.example.apaga.ui.base

import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.thoughtbot.expandablerecyclerview.ExpandableRecyclerViewAdapter
import com.thoughtbot.expandablerecyclerview.models.ExpandableGroup
import com.thoughtbot.expandablerecyclerview.viewholders.ChildViewHolder
import com.thoughtbot.expandablerecyclerview.viewholders.GroupViewHolder

abstract class BaseExpandableRecyclerViewAdapter<
        DATA,
        GVH : BaseExpandableRecyclerViewAdapter.BaseGroupHolder<*>,
        CVH : BaseExpandableRecyclerViewAdapter.BaseChildHolder<*>
        >(groups: List<ExpandableGroup<*>?>?) : ExpandableRecyclerViewAdapter<GVH, CVH>(groups) {

    fun notifyGroupDataChanged() {
        expandableList.expandedGroupIndexes = BooleanArray(groups.size)
        for (i in groups.indices) {
            expandableList.expandedGroupIndexes[i] = false
        }
    }

    abstract fun setGroupData(data: DATA)

    abstract class BaseGroupHolder<G>(itemView: View) : GroupViewHolder(itemView) {
        abstract fun bindGroup(groupItem: G)
    }

    abstract class BaseChildHolder<C>(itemView: View) : ChildViewHolder(itemView) {
        abstract fun bindChild(childItem: C)
    }

}